### PR TITLE
Story 302 Customize section titles on account page

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
* Adds support to pass a title attribute to [pmpro_account] shortcode when it has a single section as param.

image
image

image

image
image
image
image
image
image

### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Extract from shortcode attributes the "title" and if it exists and meet conditions use it to render given single section. Otherwise ignore it and use default.

Resolves XXX.

### How to test the changes in this Pull Request:

1. Create a page or post
2. Add the shortcode and play with different combinations, one section, more than one, add title, make it empty, don't add it.
3. Check results.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
 Adds support to pass a title attribute to [pmpro_account] shortcode when it has a single section as a param.

### CODE REVIEW

**Comment by MaximilianoRicoTabo** [comment]
submitted as as a draft. It lacks the part regarding customize gutenberg which will go in the next commit

File: shortcodes/pmpro_account.php [hunk 1]
```diff
@@ -32,10 +36,12 @@ function pmpro_shortcode_account($atts, $content=null, $code="")
 	$invoices = $wpdb->get_results("SELECT *, UNIX_TIMESTAMP(CONVERT_TZ(timestamp, '+00:00', @@global.time_zone)) as timestamp FROM $wpdb->pmpro_membership_orders WHERE user_id = '$current_user->ID' AND status NOT IN('review', 'token', 'error') ORDER BY timestamp DESC LIMIT 6");
 	?>
 	<div id="pmpro_account">
-		<?php if(in_array('membership', $sections) || in_array('memberships', $sections)) { ?>
-			<div id="pmpro_account-membership" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_box', 'pmpro_account-membership' ) ); ?>">
+		<?php if(in_array('membership', $sections) || in_array('memberships', $sections)) {
+			$membership_title = $title_param ? $title_param : 'My Membership';
```
Comment (dparker1005) [inline shortcodes/pmpro_account.php:40]:
```suggestion
			$membership_title = $title_param ? $title_param : __( 'My Membership', 'paid-memberships-pro );

```

**Review by dparker1005 — CHANGES_REQUESTED** [review]
Looks good so far! The one note that I have is that the first parameter passed to l18n functions should always be a literal string instead of a variable. Passing a variable could interfere with the automatic parsing of the plugin files when creating translation files.

I included a sample fix of this for the Membership section. The fixes should be similar for the other three

File: shortcodes/pmpro_account.php [hunk 2]
```diff
@@ -32,10 +36,12 @@ function pmpro_shortcode_account($atts, $content=null, $code="")
...
 	<div id="pmpro_account">
-		<?php if(in_array('membership', $sections) || in_array('memberships', $sections)) { ?>
-			<div id="pmpro_account-membership" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_box', 'pmpro_account-membership' ) ); ?>">
+		<?php if(in_array('membership', $sections) || in_array('memberships', $sections)) {
+			$membership_title = $title_param ? $title_param : 'My Membership';
+		?>
 
-				<h3><?php esc_html_e("My Memberships", 'paid-memberships-pro' );?></h3>
+			<div id="pmpro_account-membership" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_box', 'pmpro_account-membership' ) ); ?>">
+				<h3><?php esc_html_e($membership_title, 'paid-memberships-pro' );?></h3>
```
[the remark this answers is not in the document]
Comment (MaximilianoRicoTabo) [reply shortcodes/pmpro_account.php]: done, thanks

**Review by dparker1005 — APPROVED** [review]

**Pull request merged**